packages/cardano-services/test/Asset/TypeormAssetProvider/TypeormAssetProvider.test.ts

```diff
@@ -0,0 +1,118 @@
+import { Cardano, ProviderError } from '@cardano-sdk/core';
+import { DbPools } from '../../../src/util';
+import { Pool } from 'pg';
+import {
+  TypeOrmNftMetadataService,
+  TypeormAssetProvider,
+  createDnsResolver,
+  getConnectionConfig,
+  getEntities
+} from '../../../src';
+import { TypeormAssetFixtureBuilder, TypeormAssetWith } from './fixtures/TypeormFixtureBuilder';
+import { clearDbPools, sleep } from '../../util';
+import { logger } from '@cardano-sdk/util-dev';
+
+describe('TypeormAssetProvider', () => {
+  let provider: TypeormAssetProvider;
+  let service: TypeOrmNftMetadataService;
+  let fixtureBuilder: TypeormAssetFixtureBuilder;
+  let dbPools: DbPools;
+
+  beforeAll(async () => {
+    const dnsResolver = createDnsResolver({ factor: 1.1, maxRetryTime: 1000 }, logger);
+    const entities = getEntities(['asset']);
+    const connectionConfig$ = getConnectionConfig(dnsResolver, 'test', 'Handle', {
+      postgresConnectionStringHandle: process.env.POSTGRES_CONNECTION_STRING_HANDLE!
+    });
+
+    service = new TypeOrmNftMetadataService({ connectionConfig$, entities, logger });
+    provider = new TypeormAssetProvider({ connectionConfig$, entities, logger, nftMetadataService: service });
+    await provider.initialize();
+    await provider.start();
+    await service.initialize();
+    await service.start();
+  });
+
+  afterAll(async () => {
+    jest.restoreAllMocks();
+
+    await provider.shutdown();
+    await service.shutdown();
+  });
+
+  beforeEach(async () => {
+    dbPools = {
+      healthCheck: new Pool({ connectionString: process.env.POSTGRES_CONNECTION_STRING_HANDLE }),
+      main: new Pool({ connectionString: process.env.POSTGRES_CONNECTION_STRING_HANDLE })
+    };
+    fixtureBuilder = new TypeormAssetFixtureBuilder(dbPools.main, logger);
+  });
+
+  afterEach(async () => {
+    await clearDbPools(dbPools);
+  });
+
+  describe('healthCheck', () => {
+    it('forwards the assetProvider health response with provider client', async () => {
+      // required a delay to determine TypeormProvider's healthCheck as healthy when subscribes to observable data source
+      const response = await provider.healthCheck();
+      expect(response).toEqual({ ok: false, reason: 'not started' });
+      while (!(await provider.healthCheck()).ok) await sleep(10);
```

Comment:
Out of scope for this PR (let's create a new ticket): when TypeormProvider `start` resolves, it should be healthy.
```suggestion
      expect(response.ok).toBe(true);
```

Reply:
Fixed on e56866e